src/Development/IDE/Core/Rules.hs

```diff
@@ -185,11 +186,24 @@ getHomeHieFile f = do
       wait <- lift $ delayedAction $ mkDelayedAction "OutOfDateHie" L.Info $ do
         hsc <- hscEnv <$> use_ GhcSession f
         pm <- use_ GetParsedModule f
-        typeCheckRuleDefinition hsc pm DoGenerateInterfaceFiles
+        source <- getSourceFileSource f
+        typeCheckRuleDefinition hsc pm DoGenerateInterfaceFiles source
       _ <- MaybeT $ liftIO $ timeout 1 wait
       ncu <- mkUpdater
       liftIO $ loadHieFile ncu hie_f
 
+getSourceFileSourceRule :: Rules ()
+getSourceFileSourceRule = define $ \GetSourceFileSource nfp -> do
```

Comment:
What is the motivation to make this a Shake rule, i.e. why should we cache the source file contents? I cannot think of any good reason

Reply:
It mainly felt right to add it to the Shake Graph. But I do not have any other motivation, I am not really familiar with the shake system, so I am happy to learn :)
I thought it would be always good to cache it.